namerd/storage/etcd/src/integration/scala/io/buoyant/namerd/storage/etcd/EtcdDtabStoreIntegrationTest.scala

```diff
@@ -0,0 +1,81 @@
+package io.buoyant.namerd.storage.etcd
+
+import com.twitter.finagle.Dtab
+import com.twitter.io.Buf
+import com.twitter.util.Activity
+import io.buoyant.etcd.{Etcd, EtcdFixture}
+import io.buoyant.namerd.DtabStore.{DtabNamespaceAlreadyExistsException, DtabNamespaceDoesNotExistException, DtabVersionMismatchException}
+import io.buoyant.namerd.{RichActivity, VersionedDtab}
+import io.buoyant.test.Exceptions
+
+class EtcdDtabStoreIntegrationTest extends EtcdFixture with Exceptions {
+
+  def mkStore(etcd: Etcd) = {
+    val key = etcd.key("/dtabs")
+    new EtcdDtabStore(key)
+  }
+
+  def extractDtab(obs: Activity[Option[VersionedDtab]]): Dtab =
+    await(obs.values.map(_.get.get.dtab).toFuture)
+
+  test("observe follows changes") { etcd =>
+    val store = mkStore(etcd)
+    val obs = store.observe("hello")
+    assert(await(obs.toFuture).isEmpty)
+    await(store.create("hello", Dtab.read("/hello => /world")))
+    assert(await(obs.toFuture).get.dtab == Dtab.read("/hello => /world"))
+    val version = await(obs.toFuture).get.version
+    await(store.update("hello", Dtab.read("/goodbye => /world"), version))
+    assert(await(obs.toFuture).get.dtab == Dtab.read("/goodbye => /world"))
+  }
```

Comment:
TIOLI: would it be worth validating that the version changes after each update as part of this test?